File: api/app/Models/Sale.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Sale extends Model
{
    protected $fillable = [
        'movement_type', 'document_type_label', 'document_number',
        'issue_date', 'series_number', 'series_prefix', 'tracking_number',
        'sale_datetime', 'branch', 'seller',
        'customer_name', 'customer_tax_id', 'customer_email',
        'customer_address', 'customer_district', 'customer_province', 'customer_department',
        'price_list', 'delivery_type', 'currency',
        'total_net', 'total_tax', 'total_gross',
        'discount_net', 'discount_gross',
        'import_source', 'import_batch', 'user_id',
    ];

    protected $casts = [
        'issue_date'   => 'date',
        'sale_datetime' => 'datetime',
        'total_net'    => 'decimal:2',
        'total_tax'    => 'decimal:2',
        'total_gross'  => 'decimal:2',
        'discount_net' => 'decimal:2',
        'discount_gross' => 'decimal:2',
    ];

    public function items(): HasMany
    {
        return $this->hasMany(SaleItem::class);
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }
}

File: api-dotnet/AdminHiitop.Api/Domain/Sales/Entities/Sale.cs
using AdminHiitop.Api.Domain.Common;

namespace AdminHiitop.Api.Domain.Sales.Entities;

public sealed class Sale : AuditableEntity
{
    public string? MovementType { get; set; }
    public string? DocumentTypeLabel { get; set; }
    public string? DocumentNumber { get; set; }
    public DateTime? IssueDate { get; set; }
    public string? SeriesNumber { get; set; }
    public string? SeriesPrefix { get; set; }
    public string? TrackingNumber { get; set; }
    public DateTime? SaleDateTime { get; set; }
    public string? Branch { get; set; }
    public string? Seller { get; set; }
    public string? CustomerName { get; set; }
    public string? CustomerTaxId { get; set; }
    public string? CustomerEmail { get; set; }
    public string? CustomerAddress { get; set; }
    public string? CustomerDistrict { get; set; }
    public string? CustomerProvince { get; set; }
    public string? CustomerDepartment { get; set; }
    public string? PriceList { get; set; }
    public string? DeliveryType { get; set; }
    public string Currency { get; set; } = "PEN";
    public decimal TotalNet { get; set; }
    public decimal TotalTax { get; set; }
    public decimal TotalGross { get; set; }
    public decimal DiscountNet { get; set; }
    public decimal DiscountGross { get; set; }
    public string? ImportSource { get; set; }
    public string? ImportBatch { get; set; }
    public int? UserId { get; set; }

    public User? User { get; set; }
    public ICollection<SaleItem> Items { get; set; } = new List<SaleItem>();
}
